specs/waku/v2/waku-v2-rpc-api.md

```diff
@@ -0,0 +1,459 @@
+---
+title: Waku v2 JSON-RPC REST API
+version: 0.0.1
+status: Raw
+authors: Hanno Cornelius <hanno@status.im>
+---
+
+# Table of Contents
+
+1. [Introduction](#introduction)
+1. [Wire Protocol](#wire-protocol)
+    1. [Transport](#transport)
+    1. [Types](#types)
+    1. [Method naming](#method-naming)
+    1. [Debug API](#debug-api)
+    1. [Relay API](#relay-api)
+    1. [Store API](#store-api)
+    1. [Filter API](#filter-api)
+1. [Example usage](#example-usage)
+1. [Changelog](#changelog)
+1. [Copyright](#copyright)
+
+# Introduction
+
+This specification describes the JSON-RPC API that Waku v2 nodes MAY adhere to. Refer to the [Waku v2 specification](https://github.com/vacp2p/specs/blob/master/specs/waku/v2/waku-v2.md) for more information on Waku v2.
+
+# Wire Protocol
+
+## Transport
+
+Nodes SHOULD expose an accessible [JSON-RPC](https://www.jsonrpc.org/specification) API. The JSON-RPC version SHOULD be `2.0`. Below is an example request:
+
+```json
+{
+  "jsonrpc":"2.0",
+  "method":"get_waku_v2_debug_info",
+  "params":[],
+  "id":1
+}
+```
+
+### Fields
+
+| Field     | Description                                         |
+| --------- | --------------------------------------------------- |
+| `jsonrpc` | Contains the used JSON-RPC version (`Default: 2.0`) |
+| `method`  | Contains the JSON-RPC method that is being called   |
+| `params`  | An array of parameters for the request              |
+| `id`      | The request ID                                      |
+
+## Types
+
+In this specification, the primitive types `Boolean`, `String`, `Number` and `Null`, as well as the structured types `Array` and `Object`, are to be interpreted according to the [JSON-RPC specification](https://www.jsonrpc.org/specification#conventions). It also adopts the same capitalisation conventions.
+
+The following structured types are defined for use throughout the document:
+
+### WakuMessage
+
+Refer to [`Waku Message` specification](https://github.com/vacp2p/specs/blob/master/specs/waku/v2/waku-message.md) for more information.
+
+`WakuMessage` is an `Object` containing the following fields:
```

Comment:
an`Object` means JSON `Object`?

Reply:
Correct. I tried to explain how these different terms are used throughout the document under the "Types" subheading. Let me know if you think it can be clarified further. 👍